### ISSUE TITLE
[ALM Accelerator - BUG]: ISSUE TITLE

### ISSUE THREAD

**Issue by Miti-Shah** [microsoft/coe-starter-kit#6971]
### Describe the issue

We have a Solution aware Power Automate flow that connects to 4 different Key vaults. For this we use 4 different connection references pointing to appropriate key vault connection.
When deploying the solution using ALM Accelerator to higher environments, in the configuration screen, we create connections in the target and map it to the appropriate connection references.
After the deployment is completed we see that all the 4 connections references are mapped to the same connection id, they are no longer mapped to the connections created and mapped in the configuration section.
This results in the solution to fail in the higher environments.

### Expected Behavior

keyvault connection should be mapping to appropriate connection references when we move the solution to higher environments

### What component are you experiencing the issue with?

ALM Accelerator Canvas App

### What solution version are you using?

1.0.20231006.2(Managed)

**Comment by RajeevPentyala** [comment]
@Miti-Shah You can utilize the toggle option (highlighted below) in the 'Connection References' tab to specify the Connection IDs. This approach will help avoid confusion when creating connections for a similar type of 'Connection Reference'.
![image](https://github.com/microsoft/coe-starter-kit/assets/104883923/1166285b-385e-4688-8f01-6a9382887f2e)

'Connection Id' can be grabbed from the browser's connection url as shown below.
![image](https://github.com/microsoft/coe-starter-kit/assets/104883923/84413b78-ef67-4a76-adf9-13baec101c1c)

**Comment by Miti-Shah** [comment]
@RajeevPentyala  we do not see the option yet under ALM , do let us know if we need to adjust any settings from our end to make it visible.

<img width="355" alt="image" src="https://github.com/microsoft/coe-starter-kit/assets/102228331/a80875f8-f1ac-4190-88eb-ae4ee4310339">

**Comment by Miti-Shah** [comment]
@RajeevPentyala we tried this method and issue still exist , we mapped respective KV connection references and closed and saved solution but issue still persists.

<img width="473" alt="image" src="https://github.com/microsoft/coe-starter-kit/assets/102228331/0180fb15-229b-4252-a151-68a4d1415e43">

**Comment by Miti-Shah** [comment]
Thanks @RajeevPentyala  also once we save and close the solution , all references shouldn't point back to 1st one. As per below screenshot
<img width="728" alt="image" src="https://github.com/microsoft/coe-starter-kit/assets/102228331/4c5fb5a8-e728-437c-b3bb-73151eb22f02">

**Comment by RajeevPentyala** [comment]
@Miti-Shah Yes, reference selection will be persisted when you close and comeback.
